mutation_reader.cc

```diff
@@ -1477,6 +1477,7 @@ class multishard_combining_reader : public flat_mutation_reader::impl {
         }
     };
 
+    const dht::sharding_info& _sharding_info;
```

Comment:
The lifetime is guaranteed to be longer than any operation because we're using sharding_info object that's static and managed by database. I agree that probably it will be better in the previous patch to also take sharding_info by const& to be consistent.

Reply:
I double-checked and sharding_info is always passed by const&. The only place that's different is repair_metadata because there we create sharding_info based on parameters from another node. It can be easily fixed by storing sharding_info as unique_ptr if we need that so in my opinion introducing inheritance later if we need it won't be a problem at all.